test/widgets/profile/profile_page_test.dart

```diff
@@ -0,0 +1,410 @@
+        expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+        await tester.pump();
+        expect(find.text(profile.username), findsNWidgets(2));
+      });
+      testWidgets('Works with object parameter', (WidgetTester tester) async {
+        await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+        expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+        await tester.pump();
+
+        expect(find.text(profile.username), findsNWidgets(2));
+      });
+    }
+  });
+  group('Profile content', () {
+    // run for both own and other profile
+    for (int i = 0; i < 2; i++) {
+      i == 0
+          ? SupabaseManager.setCurrentProfile(profile)
+          : SupabaseManager.setCurrentProfile(ProfileFactory().generateFake(id: 2));
+
+      testWidgets('avatar$i', (WidgetTester tester) async {
+        await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+        await tester.pump();
+
+        expect(find.descendant(of: find.byType(Avatar), matching: find.byKey(const Key('avatarTappable'))),
+            findsOneWidget);
+      });
+      testWidgets('username$i', (WidgetTester tester) async {
+        await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+        await tester.pump();
+
+        expect(find.text(profile.username), findsNWidgets(2));
+      });
+      testWidgets('fullName$i', (WidgetTester tester) async {
+        await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+        await tester.pump();
+
+        expect(find.text(profile.fullName), findsAtLeastNWidgets(1));
+      });
+      testWidgets('description$i', (WidgetTester tester) async {
+        await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+        await tester.pump();
+
+        expect(find.text(profile.description!), findsOneWidget);
+      });
+      testWidgets('age$i', (WidgetTester tester) async {
+        await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+        await tester.pump();
+
+        expect(find.text(profile.age.toString()), findsOneWidget);
+      });
+      testWidgets('gender$i', (WidgetTester tester) async {
+        await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+        await tester.pump();
+
+        final genderFinder = find.byKey(const Key('gender'));
+        expect(genderFinder, findsOneWidget);
+      });
+      testWidgets('features$i', (WidgetTester tester) async {
+        await pumpMaterial(tester, ProfilePage.fromProfile(profile));
+        await tester.pump();
+
+        final scrollableFinder = find.byType(Scrollable).first;
+        expect(scrollableFinder, findsOneWidget);
+
+        for (int h = 0; h < profile.features!.length; h++) {
+          final feature = profile.features![h];
+          final featureFinder = find.byKey(Key('feature_${feature.name}'));
+          await tester.scrollUntilVisible(featureFinder, 500, scrollable: scrollableFinder);
+          expect(featureFinder, findsOneWidget);
+        }
```

Comment:
can you also check that no other features are shown, i.e. check if the feature count is the profile.features.length